src/handlers.rs

```diff
@@ -411,21 +411,30 @@ fn handle_make_temp_dir(d: *const DenoC, base: &msg::Base) -> Box<Op> {
   }()))
 }
 
-fn handle_mkdir_sync(d: *const DenoC, base: &msg::Base) -> Box<Op> {
-  let msg = base.msg_as_mkdir_sync().unwrap();
-  let path = msg.path().unwrap();
+fn handle_mkdir(d: *const DenoC, base: &msg::Base) -> Box<Op> {
+  let msg = base.msg_as_mkdir().unwrap();
+  let cmd_id = base.cmd_id();
   // TODO let mode = msg.mode();
+  let path = msg.path().unwrap();
   let deno = from_c(d);
-  debug!("handle_mkdir_sync {}", path);
+
+  debug!("handle_mkdir {}", path);
+
   if !deno.flags.allow_write {
     return odd_future(permission_denied());
   }
 
-  // TODO(ry) use blocking
   Box::new(futures::future::result(|| -> OpResult {
     // TODO(ry) Use mode.
     deno_fs::mkdir(Path::new(path))?;
-    Ok(None)
+    let builder = &mut FlatBufferBuilder::new();
```

Comment:
this probably is not right, please let me know how should it be refactored.

Reply:
`Ok(None)` doesn't work?

Reply:
that's what I'm not sure about.
I've tried `Ok(None)` first but this error thrown:

```
Expecting promise in table. 0
```

https://github.com/denoland/deno/blob/35bc9ddf636734a424b8f69ca7a49af071193002/js/dispatch.ts#L16

Reply:
seems `Ok(None)` does not return back `cmd_id` to js, I thought it's related to `// TODO(ry) use blocking`, so i've come up with this solution.